WalletWasabi.Backend/Config.cs

```diff
@@ -193,5 +167,58 @@ public void AssertFilePathSet()
 				throw new NotSupportedException($"{nameof(FilePath)} is not set. Use {nameof(SetFilePath)} to set it.");
 			}
 		}
+
+		private bool TryEnsureBackwardsCompatibility(string jsonString)
+		{
+			var jsObject = JsonConvert.DeserializeObject<JObject>(jsonString);
+			bool saveIt = false;
+
+			if (jsObject.TryGetValue("MainNetBitcoinCoreHost", out JToken jMainNetBitcoinCoreHost))
+			{
+				int port = Constants.DefaultMainNetBintcoinP2pPort;
+				if (jsObject.TryGetValue("MainNetBitcoinCorePort", out JToken jMainNetBitcoinCorePort) && int.TryParse(jMainNetBitcoinCorePort.ToString(), out int p))
```

Comment:
This is repeated many times and I think it could be simplified with an extension methos `JObject.ValueOrDefault<T>()`:
```c#
port = jsObject.ValueOrDefault<ushort>("MainNetBitcoinCorePort", Constants.DefaultMainNetBintcoinP2pPort);
```

Take into account that valid ports are ushort, in that way invalid values like -10, 999999 or any other invalid value would be replaced by the passed default value. 

This extension method would be something like this:
```c#
public static T ValueOrDefault<T>(this JObject jobj, string propertyName, T defaultValue)
{
   try{
       return jsObject.Value<T>(propertyName);
   }catch(Exception)
       return defaultValue;
   }
}
```

Reply:
`EndPointParser.TryParse` take care of these stuff, doesn't it?

Reply:
Actually the extension wasn't needed, because `jsObject.Value<string>("MainNetBitcoinCoreHost");` does exactly that. Returns null if it's not present.

https://github.com/zkSNACKs/WalletWasabi/pull/1945/commits/52e0022d486dc578ebfa060113145ebd9f7e5243